util/internal/integrationtest/backend/elasticsearch/src/main/java/org/hibernate/search/util/impl/integrationtest/backend/elasticsearch/SearchContainer.java

```diff
@@ -0,0 +1,147 @@
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 7 ) {
+				container.withEnv( "cluster.deprecation_indexing.enabled", "false" )
+						.withEnv( "xpack.profiling.enabled", "false" )
+						.withEnv( "xpack.ent_search.enabled", "false" );
+			}
+			return container.withEnv( "cluster.deprecation_indexing.enabled", "false" )
+					.withEnv( "indices.lifecycle.history_index_enabled", "false" )
+					.withEnv( "slm.history_index_enabled", "false" )
+					.withEnv( "stack.templates.enabled", "false" )
+					.withEnv( "xpack.ml.enabled", "false" )
+					.withEnv( "xpack.monitoring.templates.enabled", "false" )
+					.withEnv( "xpack.watcher.enabled", "false" );
+		}
+
+		if ( version.majorOptional().orElse( Integer.MIN_VALUE ) == 7 ) {
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 15 ) {
+				container.withEnv( "cluster.deprecation_indexing.enabled", "false" )
+						.withEnv( "slm.history_index_enabled", "false" );
+			}
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 8 ) {
+				container.withEnv( "stack.templates.enabled", "false" );
+			}
+			if ( version.minor().orElse( Integer.MAX_VALUE ) > 5 ) {
+				container.withEnv( "indices.lifecycle.history_index_enabled", "false" );
+			}
+			return container.withEnv( "xpack.ml.enabled", "false" )
+					.withEnv( "xpack.watcher.enabled", "false" );
+		}
+
+		return container;
+	}
+
+	private static GenericContainer<?> opensearch(String name, String tag) {
+		return common( name, tag )
+				.withEnv( "logger.level", "WARN" )
+				.withEnv( "discovery.type", "single-node" )
+				// Prevent swapping
+				.withEnv( "bootstrap.memory_lock", "true" )
+				// OpenSearch expects SSL and uses a self-signed certificate by default;
+				// it's not practical for testing, so we disable that.
+				.withEnv( "plugins.security.disabled", "true" )
+				// ISM floods the logs with errors, and we don't need it.
+				// See https://docs-beta.opensearch.org/im-plugin/ism/settings/
+				.withEnv( "plugins.index_state_management.enabled", "false" )
+				// Disable disk-based shard allocation thresholds: on large, relatively full disks (>90% used),
+				// it will lead to index creation to get stuck waiting for other nodes to join the cluster,
+				// which will never happen since we only have one node.
+				// See https://www.elastic.co/guide/en/elasticsearch/reference/7.17/modules-cluster.html#disk-based-shard-allocation
+				// See https://opensearch.org/docs/latest/opensearch/popular-api/#change-disk-watermarks-or-other-cluster-settings
+				.withEnv( "cluster.routing.allocation.disk.threshold_enabled", "false" );
+	}
+
+	/*
+	 * Suppress "Resource leak: '<unassigned Closeable value>' is never closed". Testcontainers take care of closing
+	 * this resource in the end.
+	 */
+	@SuppressWarnings("resource")
+	private static GenericContainer<?> common(String image, String tag) {
+		return new GenericContainer<>( DockerImageName.parse( image ).withTag( tag ) )
```

Comment:
This means we completely ignore the elasticsearch/opensearch Dockerfiles, doesn't it?

I mean, I suppose that's fair, as we will most likely update the version manually anyway, but in that case the Dockerfiles should probably be removed.

It would be nice to use the dockerfile at least for opensearch, at least by default when passing `-Dtest.elasticsearch.distribution=opensearch` or `-Dtest.elasticsearch.distribution=amazon-opensearch-service` and not passing a specific version/tag. But I guess this would make configuring the version in Hibernate Search (`hibernate.search.backend.version`) quite complex, so it's probably better not to do that.